ios/RNApplePayWrappers.m

```diff
@@ -0,0 +1,245 @@
+
+    NSDictionary *appleConfigurationDict = configuration[@"applePayConfiguration"];
+
+    NSString *merchantId = appleConfigurationDict[@"merchantId"];
+    NSString *currency = configuration[@"amount"][@"currency"];
+    NSString *countryCode = appleConfigurationDict[@"countryCode"];
+
+    NSArray *paymentSummaryItems = [RNApplePayWrappers summaryItemsFromAppleConfiguration:appleConfigurationDict];
+
+    JPApplePayConfiguration *appleConfiguration = [[JPApplePayConfiguration alloc] initWithMerchantId:merchantId
+                                                                                             currency:currency
+                                                                                          countryCode:countryCode
+                                                                                  paymentSummaryItems:paymentSummaryItems];
+
+    appleConfiguration.merchantCapabilities = [RNApplePayWrappers merchantCapabilitiesFromAppleConfiguration:appleConfigurationDict];
+    appleConfiguration.supportedCardNetworks = [RNWrappers cardNetworksFromConfiguration:configuration];
+    appleConfiguration.shippingMethods = [RNApplePayWrappers shippingMethodsFromAppleConfiguration:appleConfigurationDict];
+    appleConfiguration.shippingType = [RNApplePayWrappers shippingTypeFromDictionary:appleConfigurationDict];
+
+    NSNumber *billingFieldNumber = appleConfigurationDict[@"requiredBillingContactFields"];
+    appleConfiguration.requiredBillingContactFields = [RNApplePayWrappers contactFieldsFromContactFieldValue:billingFieldNumber.intValue];
+
+    NSNumber *shippingFieldNumber = appleConfigurationDict[@"requiredShippingContactFields"];
+    appleConfiguration.requiredShippingContactFields = [RNApplePayWrappers contactFieldsFromContactFieldValue:shippingFieldNumber.intValue];
+
+    appleConfiguration.returnedContactInfo = [RNApplePayWrappers returnedInfoFromAppleConfiguration:appleConfigurationDict];
+
+    return appleConfiguration;
+}
+
+//---------------------------------------------------
+// MARK: - Helper methods
+//---------------------------------------------------
+
++ (NSArray<JPPaymentSummaryItem *> *)summaryItemsFromAppleConfiguration:(NSDictionary *)appleConfiguration {
+
+    NSArray *summaryItemsArray = appleConfiguration[@"paymentSummaryItems"];
+    NSMutableArray<JPPaymentSummaryItem *> *summaryItems = [NSMutableArray new];
+
+    for (NSDictionary *summaryItemDict in summaryItemsArray) {
+        JPPaymentSummaryItem *item = [self summaryItemFromDictionary:summaryItemDict];
+        [summaryItems addObject:item];
+    }
+
+    return summaryItems;
+}
+
++ (JPPaymentSummaryItem *)summaryItemFromDictionary:(NSDictionary *)dictionary {
+    NSString *label = dictionary[@"label"];
+    NSString *amountString = dictionary[@"amount"];
+    NSDecimalNumber *amount = [NSDecimalNumber decimalNumberWithString:amountString];
```

Comment:
this can throw NSDecimalNumberOverflowException or NSDecimalNumberUnderflowException or NSDecimalNumberExactnessException or NSDecimalNumberDivideByZeroException we need to wrap this call in a try / catch block